index.js

```diff
@@ -0,0 +1,43 @@
+import {
+  NativeEventEmitter,
+  NativeModules,
+  Platform,
+} from 'react-native';
+
+const { InAppUtils } = NativeModules;
+
+const InAppUtilsEmitter = new NativeEventEmitter(InAppUtils);
+
+const promisify = fn => (...args) => new Promise((resolve, reject) => {
```

Comment:
wdyt about using `RCTPromiseResolveBlock` on objective-c side? it's going to be a breaking change tho. wdyt?

Reply:
Personally I prefer using promises but others might still want to use callbacks. Doing it js side gives you the option to use both. Are there any advantages using `RCTPromiseResolveBlock`?

Reply:
I'd vote to just standardize on Promises, using `RCTPromiseResolveBlock`.  Better to have a single, simple API than two options.